### PR TITLE
[docs] Add Lovász number section to simple SDP examples.

### PR DESCRIPTION
**Pull request by jd-foster**
A classic SDP example:
https://en.wikipedia.org/wiki/Lov%C3%A1sz_number#Lov%C3%A1sz_%22sandwich_theorem%22

### CODE REVIEW

**Comment by blegat** [comment]
The problem is in geometric conic form so `Dualization.dual_optimizer(SCS.Optimizer)` should be used instead.

**Comment by odow** [comment]
Preview: https://jump.dev/JuMP.jl/previews/PR3399/tutorials/conic/simple_examples/#Lovász-numbers

> The problem is in geometric conic form so Dualization.dual_optimizer(SCS.Optimizer) should be used instead.

I think adding `Dualization` to the JuMP documentation is a separate PR that probably deserves its own tutorial.

File: docs/src/tutorials/conic/simple_examples.jl [hunk 1]
```diff
@@ -306,6 +306,86 @@ end
...
+# Let ``i, \, j`` be integers such that ``1 \leq i < j \leq 5``.
+# We define ``A^{ij}`` to be the ``5 \times 5`` symmetric
+# matrix with entries ``(i,j)`` and ``(j,i)`` equal to 1, with all other entries 0.
+# Let  ``E`` be the graph's edge set; in this example, ``E`` contains
+# (1,2), (2,3), (3,4), (4,5), (5,1)
+# and their transposes. The Lovász number can be computed from the program 
+# ```math
+# \begin{align}
+#     \text{max}      & \quad    J • X & \\
+#     \text{    s.t.} & \quad A^{ij} • X = 0 \text{ for all } (i,j) \notin E   \\
```
Comment (odow) [inline docs/src/tutorials/conic/simple_examples.jl:341]:
```suggestion
#     \text{    s.t.} & \quad A^{ij} • X = 0 \quad \forall (i,j) \notin E   \\
```

File: docs/src/tutorials/conic/simple_examples.jl [hunk 2]
```diff
@@ -306,6 +306,86 @@ end
...
+            if !((i, j) in E || (j, i) in E)
+                A = zeros(Int, 5, 5)
+                A[i, j] = 1
+                A[j, i] = 1
+                @constraint(model, LinearAlgebra.dot(A, X) == 0)
+            end
+        end
+    end
+    @constraint(model, LinearAlgebra.tr(LinearAlgebra.I * X) == 1)
+
```
Comment (odow) [inline docs/src/tutorials/conic/simple_examples.jl:376]:
```suggestion
```

File: docs/src/tutorials/conic/simple_examples.jl [hunk 3]
```diff
@@ -306,6 +306,86 @@ end
...
+# [_The sandwich theorem_](https://doi.org/10.37236%2F1193), 
+# Electronic Journal of Combinatorics, Volume 1, Issue 1, A1.
+
+function example_theta_problem()
+    model = Model(SCS.Optimizer)
+    set_silent(model)
+    E = [(1, 2), (2, 3), (3, 4), (4, 5), (5, 1)]
+
+    @variable(model, X[1:5, 1:5], PSD)
+
```
Comment (odow) [inline docs/src/tutorials/conic/simple_examples.jl:364]:
```suggestion
```

File: docs/src/tutorials/conic/simple_examples.jl [hunk 4]
```diff
@@ -306,6 +306,86 @@ end
...
+#
+# Knuth, D. E. (1994), 
+# [_The sandwich theorem_](https://doi.org/10.37236%2F1193), 
+# Electronic Journal of Combinatorics, Volume 1, Issue 1, A1.
+
+function example_theta_problem()
+    model = Model(SCS.Optimizer)
+    set_silent(model)
+    E = [(1, 2), (2, 3), (3, 4), (4, 5), (5, 1)]
+
```
Comment (odow) [inline docs/src/tutorials/conic/simple_examples.jl:362]:
```suggestion
```

**Review by odow — COMMENTED** [review]
LGTM. Style suggestions are optional. For these simple examples I tried to go with the fewest lines of code.

**Comment by jd-foster** [comment]
It seems (ironically?) the OPF example is failing and killing the documentation build here...

**Comment by blegat** [comment]
> I think adding `Dualization` to the JuMP documentation is a separate PR that probably deserves its own tutorial.

It does deserve it's own tutorial but we should also always use it in conic tutorials when appropriate. Especially when it's purely geometric conic form or geometric standard form.

**Comment by joaquimg** [comment]
Yeah,
I agree we should strive to use what we would recommend in the tutorials.
We can either formulate in the way the solver wants the problem or use this tutorial to introduce dualization.
We certainly dont want users complaining things are slow because they followed the tutorial example.

**Comment by odow** [comment]
> We certainly dont want users complaining things are slow because they followed the tutorial example.

It's a 5x5 PSD. They aren't going to complain it's slow. They might equally wonder why we need Dualization in this example. The current version is fine for the "simple examples" tutorial. Dualization can be a separate thing in which we actually explain the different conic formulations.

**Comment by blegat** [comment]
> It's a 5x5 PSD. They aren't going to complain it's slow.

They are going to copy-paste this and apply it to a larger scale example.

> Dualization can be a separate thing in which we actually explain the different conic formulations.

If we only talk about Dualization in one tutorial it's not enough, we should use in every conic tutorials when it a clear win.

**Comment by odow** [comment]
If we want to add dualization to various conic tutorials we can add it once it has a standalone tutorial. We don't need to do it now or in this PR.

**Comment by joaquimg** [comment]
I would be fine not adding Dualization here, but, in this case, I think we should use the right formulation for the solver and highlight it.

**Comment by jd-foster** [comment]
How do you determine the right formulation for a given solver? (I genuinely don't know.)

**Comment by blegat** [comment]
> How do you determine the right formulation for a given solver? (I genuinely don't know.)

https://jump.dev/Dualization.jl/stable/manual/#DualOptimizer-1

> I would be fine not adding Dualization here, but, in this case, I think we should use the right formulation for the solver and highlight it.

Do you mean manually dualizing or manually adding bridging to free variables and vector affine conic constraints ?

> If we want to add dualization to various conic tutorials we can add it once it has a standalone tutorial. We don't need to do it now or in this PR.

I mentioned this example in the issue: https://github.com/jump-dev/JuMP.jl/issues/2348#issuecomment-1575556301 to make sure we don't forget.

**Comment by odow** [comment]
> How do you determine the right formulation for a given solver? (I genuinely don't know.)

:+1: This is my point on why we need a separate Dualization tutorial. The nuance is totally lost on 99% of JuMP users. Using Dualization in a few places is confusing. It's really an extra-for-experts option.

**Comment by blegat** [comment]
> It's really an extra-for-experts option.

It shouldn't be an option for expert though. Everyone conic user should use it and we should try to educate users to do so

**Pull request merged**